be/dtm0_log.h

```diff
@@ -0,0 +1,522 @@
+                                       struct m0_be_tx_credit       *accum);
+
+/**
+ * This routine is used to create a dtm0 persistent log.
+ *
+ * @pre tx != NULL
+ * @pre seg != NULL
+ * @post None
+ *
+ * @param tx handle to an already opened transaction.
+ * @param seg be segment on which to create the log.
+ * @param out variable in which to store the pointer to the newly created
+ *        log.
+ * @return 0 on success, anything else is a failure.
+ */
+M0_INTERNAL int m0_be_dtm0_log_create(struct m0_be_tx        *tx,
+                                      struct m0_be_seg       *seg,
+                                      struct m0_be_dtm0_log **out);
+
+/**
+ * This routine is used to destroy a previously created persistent dtm0 log
+ *
+ * @pre tx != NULL
+ * @pre *log != NULL
+ * @post None
+ *
+ * @param tx handle to an already opened transaction.
+ * @param log Pointer to the dtm0 log which we wish to destroy
+ * @return None
+ */
+M0_INTERNAL void m0_be_dtm0_log_destroy(struct m0_be_tx        *tx,
+                                        struct m0_be_dtm0_log **log);
+
+/**
+ * This routine is used to update a record in the dtm0 log. If the record
+ * does not exist it will be inserted.
+ *
+ * @pre payload != NULL
+ * @pre m0_be_dtm0_log__invariant(log)
+ * @pre m0_dtm0_tx_desc__invariant(txd)
+ * @pre m0_mutex_is_locked(&log->dl_lock)
+ * @post None
+ *
+ * @param log Pointer to the dtm0 log which we wish to destroy
+ * @param tx handle to an already opened transaction.
+ * @param txd A valid dtm0 tx descriptor that we want to store in the log
+ *        record.
+ * @param payload The payload is an opaque structure for dtm0 log to store
+ *        in the log record.
+ * @return None
+ */
+M0_INTERNAL int m0_be_dtm0_log_update(struct m0_be_dtm0_log  *log,
+                                      struct m0_be_tx        *tx,
+                                      struct m0_dtm0_tx_desc *txd,
+                                      struct m0_buf          *payload);
+
+/**
+ * Given a pointer to a dtm0 transaction id, this routine searches the
+ * log for the log record matching this tx id and returns it.
+ *
+ * @pre m0_be_dtm0_log__invariant(log)
+ * @pre m0_dtm0_tid__invariant(id))
+ * @pre m0_mutex_is_locked(&log->dl_lock)
+ *
+ * @post None
+ *
+ * @param log Pointer to the dtm0 log which we wish to destroy
+ * @param tid Pointer to a dtm0 transaction id corresponding to the record
+ *        that we wish to fetch from the log.
+ * @return m0_dtm0_log_rec corresponding to the input tid or NULL if no
+ *        such record is found.
+ */
+M0_INTERNAL
+struct m0_dtm0_log_rec *m0_be_dtm0_log_find(struct m0_be_dtm0_log    *log,
+                                            const struct m0_dtm0_tid *id);
+
+/**
+ * Given a pointer to a dtm0 transaction id, this routine searches the
+ * log for the log record matching this tx id and removes it from the log.
+ *
+ * @pre *log->dl_is_persistent == false
```

Comment:
@pre log->dl_is_persistent is false. (This is a volatile log)

Reply:
Done.